### PR TITLE
feat(Progressive Disclosure): Add the component to Patternfly-3

### PR DESCRIPTION
**Pull request by boaz0**
affects: patternfly3-react-lerna-root, patternfly-react

- [x] single line code
- [x] multi line code
- [x] tests
- [x] documentation
- [x] separate files by components
- [x] storybook: [current work](https://boaz1337.github.io/patternfly-react/?selectedKind=patternfly-react%2FForms%20and%20Controls%2FProgressive%20Disclosure&selectedStory=Single%20Level&full=0&addons=1&stories=1&panelRight=0&addonPanel=storybooks%2Fstorybook-addon-knobs)

### CODE REVIEW

File: packages/patternfly-3/patternfly-react/src/components/ProgressiveDisclosure/ProgressiveDisclosureItem.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+export const ProgressiveDisclosureItem = ({ children, show, controlButton }) => (
+  <React.Fragment>
+    {controlButton}
+    <div style={{ display: show ? '' : 'none', paddingLeft: '10px' }}>{children}</div>
```
[the remark this answers is not in the document]
Comment (boaz1337) [reply packages/patternfly-3/patternfly-react/src/components/ProgressiveDisclosure/ProgressiveDisclosureItem.js]: Um, for some reason, the css file doesn't include my class. Maybe you have an idea why?

File: packages/patternfly-3/patternfly-react/sass/patternfly-react/_progressivedisclosure.scss [hunk 2]
```diff
@@ -0,0 +1,3 @@
+.progressive-disclosure-item-children {
+  padding-left: 5px;
+}
```
[the remark this answers is not in the document]
Comment (boaz1337) [reply packages/patternfly-3/patternfly-react/sass/patternfly-react/_progressivedisclosure.scss]: Thanks! :+1:

**Comment by boaz1337** [comment]
updated storybook: [here](https://boaz1337.github.io/patternfly-react/?selectedKind=patternfly-react%2FForms%20and%20Controls%2FProgressive%20Disclosure&selectedStory=within%20Modal&full=0&addons=1&stories=1&panelRight=0&addonPanel=storybooks%2Fstorybook-addon-knobs)

File: packages/patternfly-3/patternfly-react/src/components/ProgressiveDisclosure/ProgressiveDisclosureItem.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+export const ProgressiveDisclosureItem = ({ children, show, controlButton, itemProps }) => (
```
Comment (jeff-phillips-18) [inline packages/patternfly-3/patternfly-react/src/components/ProgressiveDisclosure/ProgressiveDisclosureItem.js]:
```suggestion
export const ProgressiveDisclosureItem = ({ children, show, controlButton, ...otherProps }) => (
```

File: packages/patternfly-3/patternfly-react/src/components/ProgressiveDisclosure/ProgressiveDisclosureItem.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+  <div {...itemProps}>
```
Comment (jeff-phillips-18) [inline packages/patternfly-3/patternfly-react/src/components/ProgressiveDisclosure/ProgressiveDisclosureItem.js]:
```suggestion
  <div {...otherProps}>
```

File: packages/patternfly-3/patternfly-react/src/components/ProgressiveDisclosure/ProgressiveDisclosureItem.js [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+    {controlButton}
+    <div className={`progressive-disclosure-item-children ${show ? 'show' : 'hidden'}`}>{children}</div>
+  </div>
+);
+
+ProgressiveDisclosureItem.propTypes = {
+  /** progressive disclosure item control button: see more at ProgressiveDisclosure */
+  controlButton: PropTypes.node,
+  /** progressive disclosure item wrapper div props */
+  itemProps: PropTypes.object,
```
Comment (jeff-phillips-18) [inline packages/patternfly-3/patternfly-react/src/components/ProgressiveDisclosure/ProgressiveDisclosureItem.js]: Remove this prop

**Comment by jeff-phillips-18** [comment]
@patternfly/patternfly-react-ux

**Comment by serenamarie125** [comment]
@boaz0 i had reviewed this in another PR - how can we test?  The link doesn't seem to work for me

**Comment by boaz0** [comment]
Hi @serenamarie125 
Here is the new link: [storybook](https://644-pr-patternfly-react-patternfly.surge.sh/patternfly-3/index.html?selectedKind=patternfly-react%2FForms%20and%20Controls%2FProgressive%20Disclosure&selectedStory=Single%20Level&full=0&addons=1&stories=1&panelRight=0&addonPanel=storybooks%2Fstorybook-addon-knobs)

**Comment by boaz0** [comment]
ping @jeff-phillips-18 @serenamarie125

**Comment by boaz0** [comment]
Umm that's odd. I did set the margin to be `20px` but the preview still sets it to `5px`.

File: packages/patternfly-3/patternfly-react/less/progressivedisclosure.less [hunk 6]
```diff
@@ -0,0 +1,3 @@
+.progressive-disclosure-item-children {
```
Comment (jeff-phillips-18) [inline packages/patternfly-3/patternfly-react/less/progressivedisclosure.less]: should have caught this before, sorry. Could you rename to progressive-disclosure-item-pf-children ?

**Comment by jeff-phillips-18** [comment]
@serenamarie125 Running locally shows the correct indentation:
image

**Comment by dtaylor113** [comment]
Hi @boaz0, thanks again for contributing.  I just noticed in the 'with Dropdown' example, that the dropdown isn't styled like the pf-react dropdown:

progressive-disclosure

.VS.

image

**Comment by dtaylor113** [comment]
Another nit is that when launching the 'within Modal' example, there doesn't appear to be a way to dismiss the modal.  Clicking 'X' in the title, Cancel, or Save buttons do not dismiss the modal.

Other than these two issues, LGTM!  -thanks again.

**Comment by dlabaj** [comment]
@boaz0 We are going to close this due to inactivity ... if you would like to reopen it please feel free too.

**Pull request closed without merging**